connections/README.md

```diff
@@ -21,122 +22,164 @@ and spec status.
+    - [Connection Establishment](#connection-establishment)
+        - [Core Lifecycle](#core-lifecycle)
+        - [Bootstrap Protocol Negociation](#bootstrap-protocol-negociation)
+        - [Advanced Concepts and Manipulation](#advanced-concepts-and-manipulation)
+            - [Extended Protocol Negociation](#extended-protocol-negociation)
+            - [Securing Connections](#securing-connections)
+            - [Multiplexing Streams](#multiplexing-streams)
+        - [Upgrading](#upgrading)
     - [Practical Considerations](#practical-considerations)
-        - [Interoperability](#interoperability)
         - [State Management](#state-management)
             - [Peer Metadata Storage](#peer-metadata-storage)
             - [Connection Limits](#connection-limits)
         - [Connection Lifecycle Events](#connection-lifecycle-events)
     - [Future Work](#future-work)
+    - [Miscellaneous](#miscellaneous)
+        - [A Note about Interoperability](#a-note-about-interoperability)
+        - [Protocol IDs (Extended)](#protocol-ids-extended)
+    
 
 ## Overview
 
-This document describes the process of establishing connections to new peers in
-libp2p and, if necessary, adding security and stream multiplexing capabilities
-to "raw" connections provided by transport protocols.
+This document describes the process and anatomy of establishing connections to new peers in
+libp2p, alongside describing several fundamental aiding protocols and processes. 
 
-We also discuss opening new streams over an existing connection, and the
-protocol negotiation process that occurs to route traffic to the correct
-protocol handler.
+We also discuss the procol negociation processes that help set up sessions, and opening new additional connections over an existing connection.
 
-This document does not cover the establishment of "transport level" connections,
+This document does not cover the establishment of "transport level" connections, 
 for example opening "raw" TCP sockets, as those semantics are specific to each
 transport.
 
 What is covered here is the process that occurs after making the initial
-transport level connection, up to the point where "application level" streams
-are opened, and their protocols are identified and data is routed appropriately
+transport level connection, up to the point where "application level" connections
+are opened, and their protocols are identified, and data is routed appropriately
 to handler functions.
 
-
 ## Definitions
 
-A **connection** is a reliable, bidirectional communication channel between two
-libp2p peers that provides **security** and the ability to open multiple
-logically independent **streams**.
+### Protocol
+
+Protocols lay at the heart of libp2p, providing modularity by availability on both parties.
+
+Protocols are abritrary processes on both ends of the libp2p [session] that write/read to the underlying [connection].
```

Comment:
A protocol doesn't need to be a process.
Tiny typo:

```suggestion
Protocols are arbitrary processes on both ends of the libp2p [session] that write/read to the underlying [connection].
```

Reply:
They do not read/write to the underlying connection, but to a stream.

Reply:
> They do not read/write to the underlying connection, but to a stream.

They do, see my comment above, this document should not be specific to application framework structure, only explain the fundamentals of connection establishment.